### PR TITLE
Daily Dev Bumps

### PR DESCRIPTION
**Pull request by CoderDake**
# Description
This workflow will run a dev bump every day  at midnight.

For now, it uses my email as the author. In the future we will want to create a github devtools account that can be the author instead. This issue is tracked here: https://github.com/flutter/devtools/issues/4949

# Example Auto Commit

The following change is what an auto dev bump will look like
https://github.com/flutter/devtools/commit/76aff6b892ca980f5808f2ccadc77425d40ab0d5

**QUESTION:** Should we remove the changelog entry when doing the auto commits?

# Example Failure
Doing a dev bump on a release version will fail. This is to prevent the workflow from bumping the version while we are doing a release.
Screen Shot 2022-12-14 at 9 45 23 AM

### CODE REVIEW

**Comment by kenzieschmoll** [comment]
I don't think we want to add a new CHANGELOG entry for each bump. We should either modify the existing dev entry, or ignore the CHANGELOG update all together.

File: tool/update_version.dart [hunk 1]
```diff
@@ -33,7 +34,9 @@ void main(List<String> args) async {
 }
 
 Future<void> performTheVersionUpdate(
-    {required String currentVersion, required String newVersion}) async {
+    {required String currentVersion,
+    required String newVersion,
+    bool modifyChangeLog = true}) async {
```
Comment (kenzieschmoll) [inline tool/update_version.dart:39]: nit: tc

File: tool/update_version.dart [hunk 2]
```diff
@@ -150,6 +155,7 @@ void writeVersionToVersionFile(File versionFile, String version) {
 }
 
 void writeVersionToChangelog(File changelog, String version) {
+  //TODO Do not do this for devv file
```
Comment (kenzieschmoll) [inline tool/update_version.dart:158]: is this todo necessary?

Comment (CoderDake) [reply tool/update_version.dart:158]: oops this is extraneous

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**